### PR TITLE
generate: Use go mod edit to install properly versioned tools

### PR DESCRIPTION
**Pull request by 3v1n0**
Parse the tools using jq so that we can easily get the tool path and version and install the one explicitly requested

### CODE REVIEW

File: gh-actions/go/generate/action.yaml
Comment (didrocks) [inline gh-actions/go/generate/action.yaml:30]: Please set -eu on the script, like all others, to ensure we stop early if we can’t update/install jq.

**Review by didrocks — CHANGES_REQUESTED** [review]
Thanks for the change, this is indeed an improvement! I have comment before approving it.

File: gh-actions/go/generate/action.yaml
Comment (3v1n0) [reply gh-actions/go/generate/action.yaml:30]: Oh, weird, I was guess that it was enabled by default (iirc it is in gitlab scripts...).

File: gh-actions/go/generate/action.yaml
Comment (didrocks) [inline gh-actions/go/generate/action.yaml:30]: Can you put it after the first `echo` (again as in the other steps in this action). That way, we step name will be the echo line and not `set -eu`

**Review by didrocks — CHANGES_REQUESTED** [review]

File: gh-actions/go/generate/action.yaml
Comment (3v1n0) [reply gh-actions/go/generate/action.yaml]: ok, done I wanted to protect also a potential error on `echo` :joy:

Comment (didrocks) [reply gh-actions/go/generate/action.yaml]: If echo starts to return 1, I think the script is the least of the concern on this machine! :p

**Review by didrocks — APPROVED** [review]
Thanks! Do you have rights to merge it or should I?

**Comment by 3v1n0** [comment]
Oh, I think I can once I signed the commit, I did a further tuning though: https://github.com/canonical/desktop-engineering/compare/554197c1ad683d327b381924bede9d38ed8e671e..a118a2ecf9864afc08832b7ebf353303b103b53f

**Review by didrocks — CHANGES_REQUESTED** [review]
I remove then my approval (please don't push new code after the initial review, let's keep the scope focused).

With your latest change, you are not following what is done in other steps in the same file. For the sake of consistency, they should all be changed or none, but please don’t introduce descrepancies there (and so, this is a separate PR with a global env then).

**Comment by 3v1n0** [comment]
> With your latest change, you are not following what is done in other steps in the same file.

That's the thing... I did this (and in the same commit) because such file has _not_ anything like that, reason why IMHO it made sense because there was no other code style to follow. Thus it was better doing it in the proper way.

**Review by didrocks — APPROVED** [review]
Ack, so can we expect a followup commit to align the other action with this env variable settting?

**Comment by didrocks** [comment]
if you have the merge rights, please change your commit message before merging to use `ci(go): `

**Comment by 3v1n0** [comment]
> Ack, so can we expect a followup commit to align the other action with this env variable settting?

YEah... I was thinking at it for a while...

**Pull request merged**
